packages/payload/src/admin/components/elements/Header/index.tsx

```diff
@@ -14,14 +17,66 @@ import './index.scss'
 
 const baseClass = 'app-header'
 
+const hasActions = (viewConfig): viewConfig is { actions: React.ComponentType<any>[] } => {
+  return typeof viewConfig === 'object' && viewConfig !== null && 'actions' in viewConfig
+}
+
 export const AppHeader: React.FC = () => {
   const { t } = useTranslation()
 
   const {
+    admin: {
+      components: { actions: adminActions },
+    },
     localization,
     routes: { admin: adminRoute },
   } = useConfig()
 
+  const location = useLocation()
+  const pathSegments = location.pathname.split('/')
+
+  const { collection, global } = useDocumentInfo()
+  const [collectionConfig] = useCollectionGlobalConfigs(collection?.slug, [collection?.slug])
+  const [globalConfig] = useCollectionGlobalConfigs(global?.slug, [global?.slug])
+
+  const isSanitizedConfig = (config) => config?.admin?.components?.views
+
+  const getActions = (config, pathType) => {
```

Comment:
I would move all of this out to a provider component that renders only the `actions` it is being told to render and set the actions from the different view components.